pkg/api/v1/atlasdatabaseuser_types.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2020 MongoDB.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/status"
+)
+
+// Important:
+// The procedure working with this file:
+// 1. Edit the file
+// 1. Run "make generate" to regenerate code
+// 2. Run "make manifests" to regenerate the CRD
+
+func init() {
+	SchemeBuilder.Register(&AtlasProject{}, &AtlasProjectList{})
+}
+
+type ScopeType string
+
+const (
+	ClusterScopeType  ScopeType = "CLUSTER"
+	DataLakeScopeType ScopeType = "DATA_LAKE"
+)
+
+// AtlasDatabaseUserSpec defines the desired state of Database User in Atlas
+type AtlasDatabaseUserSpec struct {
+	// Project is a reference to AtlasProject resource the user belongs to
+	Project ResourceRef `json:"projectRef"`
+
+	// DatabaseName is a Database against which Atlas authenticates the user. Must be equal to 'admin'
+	// as only SCRAM authentication is supported so far.
+	DatabaseName string `json:"databaseName"`
+
+	// DeleteAfterDate is a timestamp in ISO 8601 date and time format in UTC after which Atlas deletes the user.
+	// The specified date must be in the future and within one week of the time you make the API request.
+	DeleteAfterDate string `json:"deleteAfterDate,omitempty"`
+
+	// Labels is an array containing key-value pairs that tag and categorize the database user.
+	// Each key and value has a maximum length of 255 characters.
+	Labels []LabelSpec `json:"labels,omitempty"`
+
+	// Roles is an array of this user's roles and the databases / collections on which the roles apply. A role allows
+	// the user to perform particular actions on the specified database.
+	Roles []RoleSpec `json:"roles"`
```

Comment:
What's the default behaviour if this list is left empty?

Reply:
CRD validation will pass, now I'm thinking that we need to add validation for the list shouldn't be empty.

Reply:
I mean Atlas-side -- what will Atlas do if this list is left empty?

Reply:
I'm also interested in this, I know the agent does not handle a null value for users. I believe it will correctly. create a user with no roles but it would be good to be able to confirm this!

Reply:
The Atlas returns an error if Roles is empty:
```
{"detail":"The required attribute roles was not specified.","error":400,"errorCode":"MISSING_ATTRIBUTE","parameters":["roles"],"reason":"Bad Request"}
```

Reply:
ack thanks